In [1]:
from math import sqrt
import warnings
warnings.filterwarnings('ignore')

#### Задача 1

Случайная непрерывная величина A имеет равномерное распределение на промежутке (200, 800].
Найдите ее среднее значение и дисперсию.

In [2]:
a = 200
b = 800

Математическое ожидание равномерно распределеной случайной величины равно среднему арифметическому

In [3]:
m = (a + b) / 2
m

500.0

Дисперсия

In [4]:
v = (b - a)**2 / 12
v

30000.0

#### Задача 2

О случайной непрерывной равномерно распределенной величине B известно, что ее дисперсия равна 0.2.
Можно ли найти правую границу величины B и ее среднее значение зная, что левая граница равна 0.5?
Если да, найдите ее.

Формулы математического ожидания и дисперсии для равномерного распределения<br>
m = (a + b) / 2 <br>
v = (b - a) ** 2 / 12

Известно

In [5]:
a = 0.5
v = 0.2

Подставим эти значения в формулу дисперсии и найдем b

0.2 = (b - 0.5) ** 2 / 12

2.4 = (b - 0.5) ** 2

2.4 = b ** 2 - b + 0.25

b ** 2 - b - 2.15 = 0

In [6]:
D = 1 + 4*1*2.15
D

9.6

In [7]:
d = sqrt(D)
d

3.0983866769659336

In [8]:
b1 = (1 - d) / 2
b1

-1.0491933384829668

In [9]:
b2 = (1 + d) / 2
b2

2.049193338482967

Значение b1 = -1.05 не подходит, так как правая граница не может быть меньше левой

In [10]:
b = b2
b

2.049193338482967

Математическое ожидание

In [11]:
m = (a + b) / 2
m

1.2745966692414834

#### Задача 3

Непрерывная случайная величина X распределена нормально и задана плотностью распределения
f(x) = (1 / (4 * sqrt(2pi))) * exp((-(x+2)**2) / 32)
Найдите:
а). M(X)
б). D(X)
в). std(X) (среднее квадратичное отклонение)

Исходя из формулы плотности распределения имеем

In [12]:
std = 4 # среднее квадратическое отклонение
variance = 16 # дисперсия
m = -2 # математическое ожидание

#### Задача 4

Рост взрослого населения города X имеет нормальное распределение.<br>
Причем, средний рост равен 174 см, а среднее квадратичное отклонение равно 8 см.<br>
Какова вероятность того, что случайным образом выбранный взрослый человек имеет рост:<br>
а). больше 182 см<br>
б). больше 190 см<br>
в). от 166 см до 190 см<br>
г). от 166 см до 182 см<br>
д). от 158 см до 190 см<br>
е). не выше 150 см или не ниже 190 см<br>
ё). не выше 150 см или не ниже 198 см<br>
ж). ниже 166 см.<br>

In [13]:
m = 174 # Математическое ожидание

In [14]:
std = 8 # среднее квадратичное отклонение

Если привести распределение к нормальному, то можно заметить, что значения роста 182 = 1, 190 = 2, 198 = 3, 166 = -1, 158 = -2, 150 - 3 

Воспользуемся значениями Z-таблицы для расчета вероятностей.

In [15]:
from scipy.stats import norm

а). больше 182 см (то есть больше 1 сигма)

In [16]:
p1 = 1 - norm.cdf(1)
p1

0.15865525393145707

б). больше 190 см (то есть больше 2 сигма)

In [17]:
p2 = 1 - norm.cdf(2)
p2

0.02275013194817921

в). от 166 см до 190 см (между -1 сигма и 2 сигма)

In [18]:
p3 = norm.cdf(2) - norm.cdf(-1)
p3

0.8185946141203637

г). от 166 см до 182 см (между -1 сигма и 1 сигма)

In [19]:
p4 = norm.cdf(1) - norm.cdf(-1)
p4

0.6826894921370859

д). от 158 см до 190 см (между -2 сигма и 2 сигма)

In [20]:
p5 = norm.cdf(2) - norm.cdf(-2)
p5

0.9544997361036416

е). не выше 150 см или не ниже 190 см (меньше -3 сигма или больше 2 сигма)

In [21]:
p6 = norm.cdf(-3) + (1 - norm.cdf(2))
p6

0.0241000299798093

ё). не выше 150 см или не ниже 198 см (меньше -3 сигма или больше 3 сигма)

In [22]:
p6 = norm.cdf(-3) + (1 - norm.cdf(3))
p6

0.0026997960632601965

ж). ниже 166 см. (меньше -1 сигма)

In [23]:
p7 = norm.cdf(-1)
p7

0.15865525393145707

#### Задача 5

На сколько сигм (средних квадратичных отклонений) отклоняется рост человека, равный 190 см, от
математического ожидания роста в популяции, в которой M(X) = 178 см и D(X) = 25 кв.см?

Переведем число 190 в значение в Z-распределении. Это и будет искомая сигма. std = 5

In [24]:
sigma = (190 - 178) / 5
sigma

2.4